src/Core/src/Platform/Android/KeyboardManager.cs

```diff
@@ -35,12 +37,34 @@ internal static void ShowKeyboard(this TextView inputView)
 			if (inputView?.Context is null)
 				throw new ArgumentNullException(nameof(inputView) + " must be set before the keyboard can be shown.");
 
-			using (var inputMethodManager = inputView.Context.GetSystemService(Context.InputMethodService) as InputMethodManager)
+			if (inputView.IsFocused)
+				Show();
+			else
 			{
-				// The zero value for the second parameter comes from 
-				// https://developer.android.com/reference/android/view/inputmethod/InputMethodManager#showSoftInput(android.view.View,%20int)
-				// Apparently there's no named value for zero in this case
-				inputMethodManager?.ShowSoftInput(inputView, 0);
+				var q = Looper.MyLooper();
+				if (q != null)
+					new Handler(q).Post(RequestFocus);
+				else
+					MainThread.InvokeOnMainThreadAsync(RequestFocus);
+
+				void RequestFocus()
+				{
+					if (inputView.IsDisposed())
+						return;
+
+					Show();
+				}
+			}
+
+			void Show()
+			{
+				using (var inputMethodManager = (InputMethodManager)inputView.Context.GetSystemService(Context.InputMethodService)!)
```

Comment:
I am always a little worried about the use of a !, why is it needed here? Can you show the error we get?

Reply:
copied old code and had not corrected yet - fixed now!